shard/committee/assignment.go

```diff
@@ -160,6 +168,28 @@ func prepareOrders(
 	return essentials, nil
 }
 
+// IsEligibleForEPoSAuction ..
+func IsEligibleForEPoSAuction(snapshot, validator *staking.ValidatorWrapper) bool {
+	if snapshot.Counters.NumBlocksToSign.Cmp(validator.Counters.NumBlocksToSign) != 0 {
```

Comment:
so we assume that snapshot of wrapper should have happened already by now, correct? That's only way these will match

Reply:
yea, snapshot for elected validator is gauranteed to exist.